src/components/menu/menu-full-screen/menu-full-screen.component.tsx

```diff
@@ -77,9 +77,15 @@ export const MenuFullscreen = ({
     }
   };
 
-  const filteredChildren = React.Children.toArray(children).filter((c) =>
-    React.isValidElement(c)
-  );
+  const flattenedChildren = React.Children.toArray(children);
+  const childrenWithMenuDividers: React.ReactNode[] = [];
+  flattenedChildren.forEach((child, index) => {
+    childrenWithMenuDividers.push(child);
+    if (index < flattenedChildren.length - 1) {
+      childrenWithMenuDividers.push(<MenuDivider />);
+    }
+  });
+  const childArray = React.Children.toArray(childrenWithMenuDividers);
```

Comment:
suggestion (non-blocking): happy for you to leave this as you have it already but we could probably reduce the amount of arrays here by using `map`

```
const childArray = React.Children.toArray(flattenedChildren.map((child, index) => {
    if (index < flattenedChildren.length - 1) {
      return (
        <>
          {child}
          <MenuDivider />
        </>
      )
    }

    return child;
  }));

Reply:
Thanks - this seems to work just as well as far as assigning keys correctly is concerned. Happy to make this change.